JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt

```diff
@@ -182,24 +142,50 @@ fun HomeScreen(
     ) { innerPadding ->
         val modifier = Modifier.padding(innerPadding)
         LoadingContent(
-            empty = posts.initialLoad,
+            empty = uiState.initialLoad,
             emptyContent = { FullScreenLoading() },
-            loading = posts.loading,
+            loading = uiState.loading,
             onRefresh = onRefreshPosts,
             content = {
                 HomeScreenErrorAndContent(
-                    posts = posts,
+                    posts = uiState.posts,
+                    isShowingErrors = uiState.errorMessages.isNotEmpty(),
                     onRefresh = {
                         onRefreshPosts()
                     },
                     navigateToArticle = navigateToArticle,
-                    favorites = favorites,
+                    favorites = uiState.favorites,
                     onToggleFavorite = onToggleFavorite,
                     modifier = modifier.supportWideScreen()
                 )
             }
         )
     }
+
+    // Process one error message at a time and show them as Snackbars in the UI
+    if (uiState.errorMessages.isNotEmpty()) {
+        val errorMessage = remember(uiState) { uiState.errorMessages[0] }
+
+        val errorMessageText: String = stringResource(errorMessage.messageId)
+        val retryMessage = stringResource(id = R.string.retry)
+
+        // If onRefreshPosts or onErrorDismiss change while the LaunchedEffect is running,
+        // don't restart the effect and use the latest lambda values.
+        val onRefreshPostsState by rememberUpdatedState(onRefreshPosts)
+        val onErrorDismissState by rememberUpdatedState(onErrorDismiss)
+
+        LaunchedEffect(errorMessage, errorMessageText, scaffoldState) {
```

Comment:
You don't need errorMessageText as a val

Reply:
You do, you cannot call `stringResource` within a `LaunchedEffect`

Reply:
and do you need `errorMessageText` in LaunchedEffect? aren't you covered by `errorMessage`?

Reply:
`errorMessage` is the id of the string to show on the screen to favor localization, etc.

Reply:
I mean as `key` of LaunchedEffect, not in the body.

Reply:
It uses `ContextLocal` underneath so I'd expect `errorMessageText` to be there